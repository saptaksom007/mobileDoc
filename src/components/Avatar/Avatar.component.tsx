import React, { Component } from 'react'

import { SafeImage } from 'react-native-safe-image'
import { View } from 'react-native'
import { Badge } from 'components/Badge/Badge.component'
import { parseName } from 'utilities/parseName'

import styles, { colors, DEFAULT_SIZE } from './Avatar.styles'

interface Props {
  name: string
  size: number
  style?: any
  url?: string
}

export class Avatar extends Component<Props> {
  static defaultProps = {
    size: DEFAULT_SIZE,
    name: 'default',
  }

  render() {
    const { name, style, size, url: uri } = this.props
    return (
      <SafeImage
        key={uri}
        style={{ width: size, height: size, borderRadius: size / 2 }}
        source={{ uri }}
        fallbackComponent={
          <View
            style={[
              styles.container,
              style,
              size && { width: size, height: size },
            ]}
          >
            <Badge
              value={`${(parseName(name).firstName || 'avatar')
                .substr(0, 1)
                .toUpperCase()}`}
              size={size}
              color={`${colors[(name.codePointAt(0) || 1) % colors.length]}`}
            />
          </View>
        }
      />
    )
  }
}
