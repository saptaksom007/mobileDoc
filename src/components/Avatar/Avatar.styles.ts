import { StyleSheet, Platform } from 'react-native'

export const DEFAULT_SIZE = Platform.select({ ios: 45, android: 42 })

export const colors = [
  '#e67e22',
  '#2ecc71',
  '#3498db',
  '#8e44ad',
  '#f1c40f',
  '#e74c3c',
  '#c0392b',
  '#1abc9c',
  '#f39c12',
  '#7f8c8d'
]

export default StyleSheet.create({
  container: {
    backgroundColor: 'transparent',
    width: DEFAULT_SIZE,
    height: DEFAULT_SIZE
  }
})
