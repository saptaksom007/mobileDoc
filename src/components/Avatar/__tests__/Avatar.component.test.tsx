import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Avatar } from 'components/Avatar/Avatar.component'

it('<Avatar/> render correctly', () => {
  const tree = renderer
    .create(<Avatar name={'test'} size={10} style={{}} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
