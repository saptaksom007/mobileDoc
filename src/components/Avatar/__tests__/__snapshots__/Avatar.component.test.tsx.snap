// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar/> render correctly 1`] = `
<Image
  onError={[Function]}
  source={
    Object {
      "uri": undefined,
    }
  }
  style={
    Object {
      "borderRadius": 5,
      "height": 10,
      "width": 10,
    }
  }
/>
`;
